github/http.go

```diff
@@ -135,7 +136,33 @@ func newHttpClient(testHost string, verbose bool) *http.Client {
 		Out:         ui.Stderr,
 		Colorized:   ui.IsTerminal(os.Stderr),
 	}
-	return &http.Client{Transport: tr}
+
+	// Implement CheckRedirect callback to fix issues with net/http.
+	fixupCheckRedirect := func(req *http.Request, via []*http.Request) error {
+		// net/http doesn't send a Host header on redirect requests.
+		// TODO: Find or file a Go bug.
+		if req.Host == "" {
+			req.Host = req.URL.Host
+		}
+
+		// Maintain headers after redirect.
+		// https://github.com/golang/go/issues/4800
+		for key, val := range via[0].Header {
+			if key == "Authorization" {
```

Comment:
We should do a case-insensitive comparison here. Also, could we maintain even this header if the redirect is to the same host?